assignment/assignment.hs

```diff
@@ -52,8 +60,16 @@ count element list = 0 -- replace me
 -- help to write it as two functions. (Hint: you only
 -- need to traverse the list once)
 
-evensEqualOdds :: [ Integer ] -> Bool
-evensEqualOdds numbers = False -- replace me
+--evensEqualOdds :: [ Integer ] -> Bool
+
+--not sure how to only traverse the list once
+evensEqualOdds numbers = (evenSum numbers == oddSum numbers)
+
+evenSum [] = 0
+evenSum (hd : scnd : rest) = hd + evenSum rest
+
+oddSum [] = 0
+oddSum (hd : scnd : rest) = scnd + oddSum rest
 
```

Comment:
That's a nice way to do it, and clearly less code that my way, which just scans once:

~~~ hs
evensEqualOdds list =
    evens == odds 
    where
        ( evens, odds ) = getSums list
        getSums :: [ Integer ] -> ( Integer, Integer)
        getSums [] = ( 0, 0)
        getSums (x1 : x2 : xs) =
            ( t1 + x1, t2 + x2)
            where 
                ( t1, t2 ) = getSums xs
~~~